pkg/util/crypto/crypto.go

```diff
@@ -54,7 +54,7 @@ func GenerateSelfSignedCert(host, certPath, keyPath string, alternateIPs []net.I
 		NotAfter:  time.Now().Add(time.Hour * 24 * 365),
 
 		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
```

Comment:
I think for that to work we'd need to split this cert up into a CA cert and the server cert. Then we'd need to build the client and server certs derived from that one. 

According to my reading of the RFC though, this type of usage is completely fine and this field is optional: http://tools.ietf.org/html/rfc5280#section-4.2.1.12

Basically, leaving it off completely allows the cert to be used for anything, but including the Server flag limits it to just ServerAuth.